client/windows/CreaturePurhaseCard.cpp

```diff
@@ -15,21 +15,43 @@
 #include "../../CCallback.h"
 #include "../CreatureCostBox.h"
 #include "QuickRecruitmentWindow.h"
+#include "../gui/CGuiHandler.h"
 
 void CreaturePurhaseCard::initButtons()
 {
 	initMaxButton();
 	initMinButton();
+	initCreatureSwitcherButton();
 }
 
 void CreaturePurhaseCard::initMaxButton()
 {
-	maxButton = std::make_shared<CButton>(Point(pos.x + 52, pos.y + 178), "iam014.def", CButton::tooltip(), std::bind(&CSlider::moveToMax,slider), SDLK_m);
+	maxButton = std::make_shared<CButton>(Point(pos.x + 52, pos.y + 178), "iam014.def", CButton::tooltip(), std::bind(&CSlider::moveToMax,slider));
 }
 
 void CreaturePurhaseCard::initMinButton()
 {
-	minButton = std::make_shared<CButton>(Point(pos.x, pos.y + 178), "iam015.def", CButton::tooltip(), std::bind(&CSlider::moveToMin,slider), SDLK_m);
+	minButton = std::make_shared<CButton>(Point(pos.x, pos.y + 178), "iam015.def", CButton::tooltip(), std::bind(&CSlider::moveToMin,slider));
+}
+
+void CreaturePurhaseCard::initCreatureSwitcherButton()
+{
+	creatureSwitcher = std::make_shared<CButton>(Point(pos.x + 20, pos.y-33), "iDv6432.def", CButton::tooltip(), [&](){ switchCreatureLevel(); });
+}
+
+void CreaturePurhaseCard::switchCreatureLevel()
+{
+	OBJECT_CONSTRUCTION_CAPTURING(ACTIVATE + DEACTIVATE + UPDATE + SHOWALL);
+	auto id = vstd::find(upgradesID, creatureOnTheCard->idNumber);
+	if(creatureOnTheCard->idNumber == upgradesID.back())
+		id = upgradesID.begin();
+	else
+		id++;
+	picture = std::make_shared<CCreaturePic>(pos.x, pos.y, VLC->creh->creatures[*id]);
+	picture->moveTo(Point(pos.x + parent->pos.x, pos.y + parent->pos.y));
+	creatureOnTheCard = VLC->creh->creatures[*id];
```

Comment:
do not use `VLC->creh->creatures` directly, use `CreatureID::toCreature()` instead.